pulumi/grapl/__main__.py

```diff
@@ -41,24 +42,37 @@ def _get_subset(inputs: NomadVars, subset: Set[str]) -> NomadVars:
     return {k: inputs[k] for k in subset}
 
 
-def grapl_core_docker_image_tags(
+def _container_images(
     artifacts: Mapping[str, str], require_artifact: bool = False
-) -> NomadVars:
-    # partial apply some repeated args
-    version_tag_alias = lambda key: version_tag(key, artifacts, require_artifact)
-
-    return dict(
-        analyzer_dispatcher_tag=version_tag_alias("analyzer-dispatcher"),
-        analyzer_executor_tag=version_tag_alias("analyzer-executor"),
-        dgraph_tag="latest",
-        engagement_creator_tag=version_tag_alias("engagement-creator"),
-        graph_merger_tag=version_tag_alias("graph-merger"),
-        graphql_endpoint_tag=version_tag_alias("graphql-endpoint"),
-        model_plugin_deployer_tag=version_tag_alias("model-plugin-deployer"),
-        node_identifier_tag=version_tag_alias("node-identifier"),
-        osquery_generator_tag=version_tag_alias("osquery-generator"),
-        sysmon_generator_tag=version_tag_alias("sysmon-generator"),
-        web_ui_tag=version_tag_alias("grapl-web-ui"),
+) -> str:
+    """
+    Build a map of "task name -> docker image URL".
+    See the .nomad file's var `container_images` for further documentation.
+    """
+    img_id_builder = DockerImageIdBuilder(config.container_repository())
```

Comment:
Is there any reason to have this logic in a bare function in `__main__.py`, rather than have it be a method on `DockerImageIdBuilder` itself? It would be nice to put all the logic together into a single place.

Reply:
Yeah, I can do that.